src/app/farming/d2farming.service.ts

```diff
@@ -91,48 +92,35 @@ class D2Farming {
 
 export const D2FarmingService = new D2Farming();
 
-// Ensure that there's one open space in each category that could
-// hold an item, so they don't go to the postmaster.
+// Move all items that are not tagged or locked.
 async function makeRoomForItems(store: D2Store, moveTokens: boolean) {
   const makeRoomBuckets = await getMakeRoomBuckets();
+  const itemInfos = rxStore.getState().inventory.itemInfos;
 
   console.log('Making room');
 
-  // If any category is full, we'll move one aside
-  let itemsToMove: D2Item[] = [];
+  const itemsToMove: D2Item[] = [];
   makeRoomBuckets.forEach((makeRoomBucket) => {
     const items = store.buckets[makeRoomBucket.id];
-    if (items.length > 0 && items.length >= makeRoomBucket.capacity) {
-      // We'll move the lowest-value item to the vault.
-      const itemToMove = _.minBy(
-        items.filter((i) => !i.equipped && !i.notransfer),
-        (i) => {
-          let value = {
-            Common: 0,
-            Uncommon: 1,
-            Rare: 2,
-            Legendary: 3,
-            Exotic: 4
-          }[i.tier];
-          // And low-stat
-          if (i.primStat) {
-            value += i.primStat.value / 1000;
-          }
-          return value;
-        }
-      );
-      if (itemToMove) {
-        itemsToMove.push(itemToMove);
+    // Shaders are a special case. Only free up one space.
+    if (makeRoomBucket.hash === 2973005342) {
+      if (items.length >= makeRoomBucket.capacity) {
+        itemsToMove.push(items[0]);
```

Comment:
You'll want to follow the same logic we had before. For example, don't attempt to move non-transferable (`notransfer`) items.

Reply:
Ah thanks. But not the legendary/blue proirty stuff right? It shouldnt matter about the rarity as I think think that really affects what people like

Reply:
That shouldn't matter, but you'll need to pre-filter candidates.